pkg/networkdevice/profile/profiledefinition/profile_definition.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package profiledefinition
+
+// DeviceMeta holds device related static metadata
+// DEPRECATED in favour of profile metadata syntax
+type DeviceMeta struct {
+	// deprecated in favour of new `ProfileDefinition.Metadata` syntax
+	Vendor string `yaml:"vendor" json:"vendor"`
+}
+
+type ProfileDefinition struct {
+	Name         string            `yaml:"name" json:"name"`
+	Description  string            `yaml:"description" json:"description"`
+	SysObjectIds StringArray       `yaml:"sysobjectid" json:"sysobjectid"`
+	Device       DeviceMeta        `yaml:"device" json:"device"` // DEPRECATED
+	Metrics      []MetricsConfig   `yaml:"metrics" json:"metrics"`
+	Metadata     MetadataConfig    `yaml:"metadata" json:"metadata"`
+	MetricTags   []MetricTagConfig `yaml:"metric_tags" json:"metric_tags"`
+	StaticTags   []string          `yaml:"static_tags" json:"static_tags"`
+	Extends      []string          `yaml:"extends" json:"extends"`
+}
+
+// DeviceProfileRcConfig represent the profile stored in remote config.
+type DeviceProfileRcConfig struct {
+	Profile ProfileDefinition `json:"profile_definition"`
+}
+
+func NewProfileDefinition() *ProfileDefinition {
```

Comment:
I think standard go style guidelines say not to repeat the package name if you don't have to - e.g. this function can just be `profiledefinition.New` instead of `profiledefinition.NewProfileDefinition()`.

Reply:
In Agent repo, I see both case, `New` alone or `NewStructureName`.

I find be second one a bit more explicit, but agree that's a debatable choice :D